pkg/activator/net/throttler_test.go

```diff
@@ -636,3 +643,141 @@ func TestPickP2C(t *testing.T) {
 		}
 	})
 }
+
+func TestPickIndices(t *testing.T) {
+	tests := []struct {
+		l            string
+		pods         int
+		acts         int
+		idx          int
+		wantB, wantE int
+	}{{
+		l:     "1 pod, 1 activator",
+		pods:  1,
+		acts:  1,
+		idx:   0,
+		wantB: 0,
+		wantE: 1,
+	}, {
+		l:     "1 pod, 2 activators, this is 0",
+		pods:  1,
+		acts:  2,
+		idx:   0,
+		wantB: 0,
+		wantE: 1,
+	}, {
+		l:     "1 pod, 2 activators, this is 1",
+		pods:  1,
+		acts:  2,
+		idx:   1,
+		wantB: 0,
+		wantE: 1,
+	}, {
+		l:     "2 pods, 3 activators, this is 1",
+		pods:  2,
+		acts:  3,
+		idx:   1,
+		wantB: 1,
+		wantE: 2,
+	}, {
+		l:     "2 pods, 3 activators, this is 2",
+		pods:  2,
+		acts:  3,
+		idx:   2,
+		wantB: 0,
+		wantE: 1,
+	}, {
+		l:     "3 pods, 3 activators, this is 2",
+		pods:  3,
+		acts:  3,
+		idx:   2,
+		wantB: 2,
+		wantE: 3,
+	}, {
+		l:     "10 pods, 3 activators this is 0",
+		pods:  10,
+		acts:  3,
+		idx:   0,
+		wantB: 0,
+		wantE: 4,
+	}, {
+		l:     "10 pods, 3 activators this is 1",
+		pods:  10,
+		acts:  3,
+		idx:   1,
+		wantB: 4,
+		wantE: 7,
+	}, {
+		l:     "10 pods, 3 activators this is 2",
+		pods:  10,
+		acts:  3,
+		idx:   2,
+		wantB: 7,
+		wantE: 10,
+	}, {
+		l:     "150 pods, 5 activators this is 0",
+		pods:  150,
+		acts:  5,
+		idx:   0,
+		wantB: 0,
+		wantE: 30,
+	}, {
+		l:     "150 pods, 5 activators this is 1",
+		pods:  150,
+		acts:  5,
+		idx:   1,
+		wantB: 30,
+		wantE: 60,
+	}, {
+		l:     "10 pods, 3 activators this is 4",
+		pods:  150,
+		acts:  5,
+		idx:   4,
+		wantB: 120,
+		wantE: 150,
+	}}
+	for _, test := range tests {
+		t.Run(test.l, func(tt *testing.T) {
+			bi, ei := pickIndices(test.pods, test.idx, test.acts)
+			if got, want := bi, test.wantB; got != want {
+				t.Errorf("BeginIndex = %d, want: %d", got, want)
+			}
+			if got, want := ei, test.wantE; got != want {
+				t.Errorf("EndIndex = %d, want: %d", got, want)
+			}
+		})
+	}
+}
+
+func TestAssignSlice(t *testing.T) {
+	trackers := []*podIPTracker{
+		&podIPTracker{
+			dest: "2",
+		},
+		&podIPTracker{
+			dest: "1",
+		},
+		&podIPTracker{
+			dest: "3",
+		},
+	}
+	t.Run("idx=-1", func(t *testing.T) {
+		got := assignSlice(trackers, -1, 1)
+		if !reflect.DeepEqual(got, trackers) {
```

Comment:
Why not

```suggestion
		if !cmp.Equals(got, trackers) {
```

+ Diff?

Reply:
private type, with private members :) cmp will panic (or I'll have to add the permit private fields)

Reply:
https://github.com/knative/pkg/blob/master/kmp/diff.go#L41

Reply:
That would return an error. I can add a cmpoption, in reality.

Reply:
OKay, went with `cmp` since you all seem to like it :-)